chimney/src/main/scala/io/scalaland/chimney/dsl/TransformerDefinition.scala

```diff
@@ -18,6 +19,18 @@ final class TransformerDefinition[From, To, C <: TransformerCfg](
     val instances: Map[(String, String), Any]
 ) {
 
+  def __refineConfig[C1 <: TransformerCfg]: TransformerDefinition[From, To, C1] =
+    this.asInstanceOf[TransformerDefinition[From, To, C1]]
+
+  def __addOverride(key: String, value: Any): TransformerDefinition[From, To, C] =
+    new TransformerDefinition(overrides.updated(key, value), instances)
+
```

Comment:
In `TransformerDefinition` there is not much advantage beside being consistent with `TransformerFDefinition`.

In `TransformerFDefinition` it's a longer story:
- these methods allow to overcome some difficulties with accessing to type tree of `F[_]`
- in case of macro, we can request `WeakTypeTag` instance for type parameters, but it doesn't really work with types with arguments (it requires kind `*`, we need to wildcard type args)
- materialization of such type tree ended up with not picking the right implicit instance of `TransformerFSupport` during tree materialization (seems that type constructed in this way was somehow broken)
- using `_addOverride` and `__refineConfig` allowed to eliminate access to `F` in a macro - it's just passed through by these operations

Unfortunately, they must be public as they are invoked by the macro-expanded code. But so far we have few public stuff in `Transformer(F)Definition/Into`. It would be nice to get rid of them, but one must propose new encoding that eliminates them.

Reply:
Why not use single TransformerDefinition for TransformerF and Trasformer (with F[_] = Id)?

Reply:
This could reduce code volume (and I will try to do this later), but I'm afraid will not address original issue with WeakTypeTags for type constructors.

I tried to look at your implementation and you may hit the same issue, because of [this](https://github.com/scalalandio/chimney/pull/145/files#diff-28b933308957ae0c4ae2f692ccf28782R92). Just try to play a bit with `Either[Seq[E], *]`.

Reply:
> Why not use single TransformerDefinition for TransformerF and Trasformer (with F[_] = Id)?

Lifted transformer (`TransformerF`) is conceptually different thing to total transformer (`Transformer`). We could possibly define `Transformer[A, B] = TransformerF[Id, A, B]` and not have duplication in DSLs. But having them separated:
- we can emit efficient code without wrapping in `.pure` for `Id`
- we can mix lifted and partial DSL operations, lifting in place when first used `withFieldConstF`/`withFieldComputedF`/etc.
- keeping current `Transformer` untouched will not force current users to rewrite their code

It seems as a reasonable tradeoff to keep these types separated. At some point later we may revisit this belief.